Native image build fails for AWS custom runtime

There is something wrong with this docker file [DockerfileAwsCustomRuntime](https://github.com/micronaut-projects/micronaut-maven-plugin/blob/master/micronaut-maven-plugin/src/main/resources/dockerfiles/DockerfileAwsCustomRuntime)

Could not install libc6-dev zlib1g-dev error

Running the command

```
mvn clean package -Dpackaging=docker-native -Dmicronaut.runtime=lambda -Pgraalvm
```
Fails with this errors
```
[INFO] Step 6/23 : RUN yum update -y && yum install -y gcc gcc-c++ libc6-dev zlib1g-dev curl bash zlib zlib-devel zlib-static zip tar gzip && yum clean all
[INFO] 
[INFO]  ---> Running in 66ede04078c2
[INFO] Failed to set locale, defaulting to C.UTF-8

[INFO] Amazon Linux 2023 repository                    6.0 MB/s |  15 MB     00:02    
[INFO] Last metadata expiration check: 0:00:02 ago on Wed Jul 26 13:07:19 2023.
[INFO] Dependencies resolved.
[INFO] Nothing to do.
[INFO] Complete!
[INFO] Failed to set locale, defaulting to C.UTF-8

[INFO] Last metadata expiration check: 0:00:02 ago on Wed Jul 26 13:07:19 2023.
[INFO] No match for argument: libc6-dev
[INFO] No match for argument: zlib1g-dev
[INFO] Package bash-5.2.15-1.amzn2023.0.2.x86_64 is already installed.
[INFO] Package zlib-1.2.11-33.amzn2023.0.4.x86_64 is already installed.
[INFO] Error: Unable to find a match: libc6-dev zlib1g-dev

[ERROR] The command '/bin/sh -c yum update -y && yum install -y gcc gcc-c++ libc6-dev zlib1g-dev curl bash zlib zlib-devel zlib-static zip tar gzip && yum clean all' returned a non-zero code: 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.938 s
[INFO] Finished at: 2023-07-26T10:07:22-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.micronaut.maven:micronaut-maven-plugin:4.0.1:docker-native (default-docker-native) on project alideals-update: Execution default-docker-native of goal io.micronaut.maven:micronaut-maven-plugin:4.0.1:docker-native failed: Could not build image: The command '/bin/sh -c yum update -y && yum install -y gcc gcc-c++ libc6-dev zlib1g-dev curl bash zlib zlib-devel zlib-static zip tar gzip && yum clean all' returned a non-zero code: 1 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

--- a/micronaut-maven-integration-tests/src/it/dockerfile-docker-native-lambda/Dockerfile.aarch64
+++ b/micronaut-maven-integration-tests/src/it/dockerfile-docker-native-lambda/Dockerfile.aarch64
@@ -1,4 +1,4 @@
-FROM amazonlinux:latest AS graalvm
+FROM amazonlinux:2 AS graalvm
 ENV LANG=en_US.UTF-8
 RUN yum update -y && yum install -y gcc gcc-c++ libc6-dev zlib1g-dev curl bash zlib zlib-devel zlib-static zip tar gzip && yum clean all
 RUN curl -4 -L https://download.oracle.com/graalvm/17/latest/graalvm-jdk-17_linux-aarch64_bin.tar.gz -o /tmp/graalvm.tar.gz \
@@ -12,7 +12,7 @@ COPY dependency/* /home/app/libs/
 COPY *.args /home/app/graalvm-native-image.args
 RUN native-image @/home/app/graalvm-native-image.args -H:Class=io.micronaut.function.aws.runtime.MicronautLambdaRuntime -H:Name=application -cp "/home/app/libs/*:/home/app/classes/"
 
-FROM amazonlinux:latest
+FROM amazonlinux:2
 WORKDIR /function
 RUN yum update -y && yum install -y zip && yum clean all
 COPY --from=graalvm /home/app/application /function/func

--- a/micronaut-maven-integration-tests/src/it/dockerfile-docker-native-lambda/Dockerfile.amd64
+++ b/micronaut-maven-integration-tests/src/it/dockerfile-docker-native-lambda/Dockerfile.amd64
@@ -1,4 +1,4 @@
-FROM amazonlinux:latest AS graalvm
+FROM amazonlinux:2 AS graalvm
 ENV LANG=en_US.UTF-8
 RUN yum update -y && yum install -y gcc gcc-c++ libc6-dev zlib1g-dev curl bash zlib zlib-devel zlib-static zip tar gzip && yum clean all
 RUN curl -4 -L https://download.oracle.com/graalvm/17/latest/graalvm-jdk-17_linux-x64_bin.tar.gz -o /tmp/graalvm.tar.gz \
@@ -12,7 +12,7 @@ COPY dependency/* /home/app/libs/
 COPY *.args /home/app/graalvm-native-image.args
 RUN native-image @/home/app/graalvm-native-image.args -H:Class=io.micronaut.function.aws.runtime.MicronautLambdaRuntime -H:Name=application -cp "/home/app/libs/*:/home/app/classes/"
 
-FROM amazonlinux:latest
+FROM amazonlinux:2
 WORKDIR /function
 RUN yum update -y && yum install -y zip && yum clean all
 COPY --from=graalvm /home/app/application /function/func

--- a/micronaut-maven-plugin/src/main/resources/dockerfiles/DockerfileAwsCustomRuntime
+++ b/micronaut-maven-plugin/src/main/resources/dockerfiles/DockerfileAwsCustomRuntime
@@ -1,4 +1,4 @@
-FROM amazonlinux:latest AS graalvm
+FROM amazonlinux:2 AS graalvm
 ENV LANG=en_US.UTF-8
 ARG GRAALVM_VERSION
 ARG GRAALVM_JVM_VERSION
@@ -16,7 +16,7 @@ COPY *.args /home/app/graalvm-native-image.args
 ARG CLASS_NAME
 RUN native-image @/home/app/graalvm-native-image.args -H:Class=${CLASS_NAME} -H:Name=application -cp "/home/app/libs/*:/home/app/classes/"
 
-FROM amazonlinux:latest
+FROM amazonlinux:2
 WORKDIR /function
 RUN yum update -y && yum install -y zip && yum clean all
 COPY --from=graalvm /home/app/application /function/func

--- a/micronaut-maven-plugin/src/site/asciidoc/examples/package.adoc
+++ b/micronaut-maven-plugin/src/site/asciidoc/examples/package.adoc
@@ -217,7 +217,7 @@ command line:
 Note that changing the base image to a totally different one than the default might break image building, since the rest
 of the build steps expect a certain base image. By default, the native images are built from an `ghcr.io/graalvm/native-image-community` image.
 
-In the case of AWS custom runtime, it starts from `amazonlinux:latest`, and this cannot be changed. Also, in this case the
+In the case of AWS custom runtime, it starts from `amazonlinux:2`, and this cannot be changed. Also, in this case the
 result is not a tagged Docker image, but a `function.zip` archive that contains the launch script and the native binary.
 Essentially, what you need to upload to AWS Lambda. Also in this case, the Micronaut Maven Plugin will detect the host
 operating system architecture (based on the `os.arch` Java system property) and will install the corresponding GraalVM

